File: server/src/models/CakeOrder.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/database.js';

const CakeOrder = sequelize.define('CakeOrder', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  order_number: {
    type: DataTypes.STRING,
    unique: true,
    allowNull: false
  },
  cake_id: {
    type: DataTypes.UUID,
    allowNull: false
  },
  user_id: {
    type: DataTypes.UUID,
    allowNull: true
  },
  customer_name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  customer_phone: {
    type: DataTypes.STRING,
    allowNull: false
  },
  customer_email: {
    type: DataTypes.STRING,
    allowNull: true
  },
  cake_name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  size: {
    type: DataTypes.STRING,
    allowNull: false
  },
  weight: {
    type: DataTypes.STRING,
    allowNull: true
  },
  decorations: {
    type: DataTypes.JSONB, // выбранные украшения
    allowNull: true
  },
  inscription: {
    type: DataTypes.STRING,
    allowNull: true
  },
  special_requests: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  delivery_date: {
    type: DataTypes.DATE,
    allowNull: false
  },
  delivery_time: {
    type: DataTypes.STRING,
    allowNull: true
  },
  delivery_address: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  price: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false
  },
  status: {
    type: DataTypes.ENUM(
      'pending',      // ожидает подтверждения
      'confirmed',    // подтвержден
      'in_progress',  // в работе
      'ready',        // готов
      'completed',    // завершен
      'cancelled'     // отменен
    ),
    defaultValue: 'pending'
  },
  payment_status: {
    type: DataTypes.ENUM('pending', 'deposit_paid', 'paid', 'refunded'),
    defaultValue: 'pending'
  },
  deposit_amount: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: true
  }
}, {
  tableName: 'cake_orders',
  timestamps: true
});

export default CakeOrder;

File: flutter/cafeteria_app/lib/models/cake_order.dart
class CakeOrderRequest {
  final int cakeId;
  final String cakeName;
  final String size;
  final String weight;
  final List<String> decorations;
  final String inscription;
  final String specialRequests;
  final String deliveryDate;
  final String deliveryTime;
  final String deliveryAddress;
  final String customerName;
  final String customerPhone;
  final String customerEmail;
  final double price;

  CakeOrderRequest({
    required this.cakeId,
    required this.cakeName,
    required this.size,
    required this.weight,
    required this.decorations,
    required this.inscription,
    required this.specialRequests,
    required this.deliveryDate,
    required this.deliveryTime,
    required this.deliveryAddress,
    required this.customerName,
    required this.customerPhone,
    required this.customerEmail,
    required this.price,
  });

  Map<String, dynamic> toJson() => {
        'cake_id': cakeId,
        'cake_name': cakeName,
        'size': size,
        'weight': weight,
        'decorations': decorations,
        'inscription': inscription,
        'special_requests': specialRequests,
        'delivery_date': deliveryDate,
        'delivery_time': deliveryTime,
        'delivery_address': deliveryAddress,
        'customer_name': customerName,
        'customer_phone': customerPhone,
        'customer_email': customerEmail,
        'price': price,
      };
}

class CakeOrderResponse {
  final int id;
  final String orderNumber;
  final String status;

  CakeOrderResponse({
    required this.id,
    required this.orderNumber,
    required this.status,
  });

  factory CakeOrderResponse.fromJson(Map<String, dynamic> json) {
    return CakeOrderResponse(
      id: json['id'] as int,
      orderNumber: json['order_number'] as String,
      status: json['status'] as String? ?? 'pending',
    );
  }
}
